### PR TITLE
linting!

### LINKED ISSUE #214: add linting to the repository
Marissa Gorlick 
May 4, 2022, 9:17 PM

Adopting eslint and prettier adding both at the same time.  Proposing going folder by folder to make changes.  Want to warn rather than fail and will clean up as we complete PRs.

We should prep this week and think about merging next week while folks are engaged in the Global All Hands.

Edit

Delete

Hamilton Ulmer 
April 27, 2022, 8:29 PM
Brendan Colloran turns out we had a ticket for this :)

### PR DESCRIPTION
**Pull request by bcolloran**
closes #214

this PR sets up linting for our repo.

I've set up the config to be warn in the case of a several of the failing rules, and for a number of the rules that only had a few failures, I've gone ahead and cleared those.

This PR touches a fair number of files, but each individual commit is pretty much limited to fixes that affect only one linting rule. My advice for reviewing this PR is to use github's "commits" tab to step through each commit one at a time, it should make the review more digestible.

Also: this PR is intentionally submited in a state such that it will fail CI (I want to make sure that works). I'll update it an make sure it passes before taking it out of "draft" state.

### CODE REVIEW

**Comment by bcolloran** [comment]
@ericpgreen2 @hamilton  @AdityaHegde -- adding all 3 of you to review this just to make sure you know about it!

**Comment by ericpgreen2** [comment]
@bcolloran, what VSCode settings do you recommend to go along with this configuration? I skimmed [this LogRocket blog post](https://blog.logrocket.com/using-prettier-eslint-automate-formatting-fixing-javascript/) and they have these couple lines to avoid conflicts between `prettier` and `eslint`, but I have studied it in detail:
```
  "editor.formatOnSave": false,
  // Runs Prettier, then ESLint
  "editor.codeActionsOnSave": [
    "source.formatDocument",
    "source.fixAll.eslint"
  ],
```

**Comment by bcolloran** [comment]
@ericpgreen2 I've never used eslint's format on save, just Prettier's, so I don't have any advice on that. Hmm, I guess I've gotten by just watching for VSCode's little red squiggles to let me know when I'm running afoul of an eslint rule, and I fix those as I go. So I guess for that strategy: make sure eslint highlighting is enabled in vsc (I have the eslint plugin installed in vsc, but I don't think I have any special config for that). If you do decide to explore the lint-on-save path, let us know how it goes -- I'd be curious to hear if you recommend it.

**Comment by bcolloran** [comment]
note: this fixes #214

File: src/lib/util/guid.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export function guidGenerator() {
-  var S4 = function () {
+  let S4 = function () {
```
Comment (AdityaHegde) [inline src/lib/util/guid.ts:2]: `let` to `const` ?
Also should be moved outside guidGenerator no?
And `function () {` to `() => {`

Comment (AdityaHegde) [reply src/lib/util/guid.ts:2]: Not in scope to fix everything in this PR

**Review by AdityaHegde — APPROVED** [review]

**Pull request merged**